packages/machine/src/instructions.ts

```diff
@@ -41,109 +41,282 @@ export enum Opcode {
-   */
-  KEY_GENERATE,
-  /**
-   * Represents all instructions. Registering for this instruction will yield
-   * all messages.
-   */
-  ALL
+  IO_WAIT
 }
 
-/**
- * Instructions executed on the intiating end of a protocol, i.e., a peer who
- * starts a protocol with a peer out of nowhere.
- */
-export const instructions = {
-  [cf.legacy.node.ActionName.UPDATE]: [
-    Opcode.STATE_TRANSITION_PROPOSE,
-    Opcode.OP_GENERATE,
-    Opcode.OP_SIGN,
-    Opcode.IO_PREPARE_SEND,
-    Opcode.IO_SEND,
-    Opcode.IO_WAIT,
-    Opcode.OP_SIGN_VALIDATE,
-    Opcode.STATE_TRANSITION_COMMIT
-  ],
-  [cf.legacy.node.ActionName.SETUP]: [
-    Opcode.STATE_TRANSITION_PROPOSE,
-    Opcode.OP_GENERATE,
-    Opcode.OP_SIGN,
-    Opcode.IO_PREPARE_SEND,
-    Opcode.IO_SEND,
-    Opcode.IO_WAIT,
-    Opcode.OP_SIGN_VALIDATE,
-    Opcode.STATE_TRANSITION_COMMIT
-  ],
-  [cf.legacy.node.ActionName.INSTALL]: [
-    Opcode.KEY_GENERATE,
-    Opcode.IO_PREPARE_SEND,
-    Opcode.IO_SEND,
-    Opcode.IO_WAIT,
-    Opcode.STATE_TRANSITION_PROPOSE,
-    Opcode.OP_GENERATE,
-    Opcode.OP_SIGN_VALIDATE,
-    Opcode.OP_SIGN,
-    Opcode.IO_PREPARE_SEND,
-    Opcode.IO_SEND,
-    Opcode.STATE_TRANSITION_COMMIT
-  ],
-  [cf.legacy.node.ActionName.UNINSTALL]: [
-    Opcode.STATE_TRANSITION_PROPOSE,
-    Opcode.OP_GENERATE,
-    Opcode.OP_SIGN,
-    Opcode.IO_PREPARE_SEND,
-    Opcode.IO_SEND,
-    Opcode.IO_WAIT,
-    Opcode.OP_SIGN_VALIDATE,
-    Opcode.STATE_TRANSITION_COMMIT
-  ]
+const swap = function(msg: ClientActionMessage) {
+  const from = msg.fromAddress;
+  const to = msg.toAddress;
+  msg.fromAddress = to;
+  msg.toAddress = from;
 };
 
-/**
- * Instructions executed on the receiving end of a protocol, i.e., by a peer
- * who did not initiate a given protocol.
- */
-export const ackInstructions = {
-  [cf.legacy.node.ActionName.UPDATE]: [
-    Opcode.STATE_TRANSITION_PROPOSE,
-    Opcode.OP_GENERATE,
-    Opcode.OP_SIGN_VALIDATE,
-    Opcode.OP_SIGN,
-    Opcode.IO_PREPARE_SEND,
-    Opcode.IO_SEND,
-    Opcode.STATE_TRANSITION_COMMIT
-  ],
-  [cf.legacy.node.ActionName.SETUP]: [
-    Opcode.STATE_TRANSITION_PROPOSE,
-    Opcode.OP_GENERATE,
-    Opcode.OP_SIGN_VALIDATE,
-    Opcode.OP_SIGN,
-    Opcode.IO_PREPARE_SEND,
-    Opcode.IO_SEND,
-    Opcode.STATE_TRANSITION_COMMIT
-  ],
-  [cf.legacy.node.ActionName.INSTALL]: [
-    Opcode.KEY_GENERATE,
-    Opcode.STATE_TRANSITION_PROPOSE,
-    Opcode.OP_GENERATE,
-    Opcode.OP_SIGN,
-    Opcode.IO_PREPARE_SEND,
-    Opcode.IO_SEND,
-    Opcode.IO_WAIT,
-    Opcode.OP_SIGN_VALIDATE,
-    Opcode.STATE_TRANSITION_COMMIT
-  ],
-  [cf.legacy.node.ActionName.UNINSTALL]: [
-    Opcode.STATE_TRANSITION_PROPOSE,
-    Opcode.OP_GENERATE,
-    Opcode.OP_SIGN_VALIDATE,
-    Opcode.OP_SIGN,
-    Opcode.IO_PREPARE_SEND,
-    Opcode.IO_SEND,
-    Opcode.STATE_TRANSITION_COMMIT
-  ]
+export const FLOWS = {
+  [cf.legacy.node.ActionName.UPDATE]: {
+    0: [
+      (message, context, node) => {
```

Comment:
IMO we shouldn't be using fat arrow syntax if we're not actually using `this`. Just use regular `function() {`

Reply:
eh, the two are completely equivalent for this function - why not use arrow syntax for everything?